web/src/features/charts/bar-breakdown/utils.ts

```diff
@@ -39,15 +38,24 @@ export function getProductionCo2Intensity(
   return storage && storage > 0 ? storageCo2Intensity : dischargeCo2Intensity;
 }
 
-export function getExchangeCo2Intensity(mode, zoneData, electricityMixMode) {
-  const exchange = (zoneData.exchange || {})[mode];
-  const exchangeCo2Intensity = (zoneData.exchangeCo2Intensities || {})[mode];
+export function getExchangeCo2Intensity(
+  zoneKey: ZoneKey,
+  zoneData: ZoneDetail,
+  electricityMixMode: Mode
+) {
+  const exchange = (zoneData.exchange || {})[zoneKey];
+  const exchangeCo2Intensity = (zoneData.exchangeCo2Intensities || {})[zoneKey];
 
   if (exchange >= 0) {
     return exchangeCo2Intensity;
   }
 
-  return getCO2IntensityByMode(zoneData, electricityMixMode);
+  // We don't use getCO2IntensityByMode in order to more easily return 0 for invalid numbers
+  if (electricityMixMode === Mode.CONSUMPTION) {
+    return zoneData.co2intensity || 0;
+  }
+
+  return zoneData.co2intensityProduction || 0;
```

Comment:
Assuming these can only be undefined or a number as indicated by the type we should use nullish coalescing.
```suggestion
    return zoneData.co2intensity ?? 0;
  }

  return zoneData.co2intensityProduction ?? 0;
```

Reply:
I was considering it, but then we wouldn't catch `NaN` 🤔

Reply:
Damn number that is not really a number 😄
Let's keep it as is then until we figure out how to deal with all the undefined, none and NaN.

Reply:
Haha yeah, it is very confusing 😅